diff -Naur 90crypt.orig/cryptroot-ask.sh 90crypt/cryptroot-ask.sh
--- 90crypt.orig/cryptroot-ask.sh	2016-07-04 00:05:52.294690900 +0000
+++ 90crypt/cryptroot-ask.sh	2016-07-04 00:12:17.646964796 +0000
@@ -65,6 +65,43 @@
 
 #
 # Open LUKS device
+MountPoint=/tmp-keydisk-mount
+KeyFileName=LinuxPassPhraseFileName
+echo "Trying to get the key from disks ..." >&2
+mkdir -p $MountPoint >&2
+modprobe vfat >/dev/null >&2
+modprobe fuse >/dev/null >&2
+for SFS in /dev/sd*; do
+echo "> Trying device:$SFS..." >&2
+mount ${SFS}1 $MountPoint -t vfat -r >&2
+ntfs-3g -o ro ${SFS}1 $MountPoint >&2
+if [ -f $MountPoint/$KeyFileName ]; then
+echo "> keyfile got..." >&2
+cp $MountPoint/$KeyFileName /tmp-keyfile
+luksfile=/tmp-keyfile
+umount $MountPoint
+break
+fi
+done
+
+MountPoint=/tmp-boot-mount
+LuksHeaderPath=luks/osluksheader
+echo "Trying to get the LUKS header from disks ..." >&2
+mkdir -p $MountPoint >&2
+for SFS in /dev/sd*; do
+echo "> Trying device:$SFS..." >&2
+mount ${SFS}1 $MountPoint -r >&2
+if [ -f $MountPoint/$LuksHeaderPath ]; then
+echo "> got luks header..." >&2
+cp $MountPoint/$LuksHeaderPath /tmp-luksheader
+luksheaderfile=/tmp-luksheader
+umount $MountPoint
+break
+fi
+done
+
+cryptsetupopts="$cryptsetupopts --header $luksheaderfile"
+
 #
 
 info "luksOpen $device $luksname $luksfile $luksoptions"
diff -Naur 90crypt.orig/module-setup.sh 90crypt/module-setup.sh
--- 90crypt.orig/module-setup.sh	2016-07-04 00:05:52.294690900 +0000
+++ 90crypt/module-setup.sh	2016-07-04 00:06:14.084819479 +0000
@@ -33,8 +33,7 @@
         [[ "${host_fs_types[$dev]}" != "crypto_LUKS" ]] && continue
 
         UUID=$(
-            blkid -u crypto -o export $dev \
-                | while read line; do
+            blkid -u crypto -o export $dev                 | while read line; do
                 [[ ${line#UUID} = $line ]] && continue
                 printf "%s" "${line#UUID=}"
                 break
@@ -67,8 +66,7 @@
             [[ $_mapper = \#* ]] && continue
             [[ $_dev ]] || continue
 
-            [[ $_dev == UUID=* ]] && \
-                _dev="/dev/disk/by-uuid/${_dev#UUID=}"
+            [[ $_dev == UUID=* ]] &&                 _dev="/dev/disk/by-uuid/${_dev#UUID=}"
 
             for _hdev in "${!host_fs_types[@]}"; do
                 [[ ${host_fs_types[$_hdev]} == "crypto_LUKS" ]] || continue
@@ -82,14 +80,7 @@
 
     inst_simple "$moddir/crypt-lib.sh" "/lib/dracut-crypt-lib.sh"
 
-    inst_multiple -o \
-        $systemdutildir/system-generators/systemd-cryptsetup-generator \
-        $systemdutildir/systemd-cryptsetup \
-        $systemdsystemunitdir/systemd-ask-password-console.path \
-        $systemdsystemunitdir/systemd-ask-password-console.service \
-        $systemdsystemunitdir/cryptsetup.target \
-        $systemdsystemunitdir/sysinit.target.wants/cryptsetup.target \
-        systemd-ask-password systemd-tty-ask-password-agent
-    inst_script "$moddir"/crypt-run-generator.sh /sbin/crypt-run-generator
+    
+    
     dracut_need_initqueue
 }
diff -Naur 90crypt.orig/parse-crypt.sh 90crypt/parse-crypt.sh
--- 90crypt.orig/parse-crypt.sh	2016-07-04 00:05:52.294690900 +0000
+++ 90crypt/parse-crypt.sh	2016-07-04 00:06:14.085819485 +0000
@@ -1,6 +1,7 @@
 #!/bin/sh
 # -*- mode: shell-script; indent-tabs-mode: nil; sh-basic-offset: 4; -*-
 # ex: ts=8 sw=4 sts=4 et filetype=sh
+DRACUT_SYSTEMD=0
 
 type crypttab_contains >/dev/null 2>&1 || . /lib/dracut-crypt-lib.sh
 
@@ -21,13 +22,12 @@
 
             luksid=${luksid##luks-}
 
-            if [ -z "$DRACUT_SYSTEMD" ]; then
+            if [ 1 ]; then
                 {
-                    printf -- 'ENV{ID_FS_TYPE}=="crypto_LUKS", '
-                    printf -- 'ENV{ID_FS_UUID}=="*%s*", ' $luksid
+                    printf -- 'ATTRS{device_id}=="?00000000-0000-*", ATTR{partition}=="ENCRYPTED_DISK_PARTITION",'
                     printf -- 'RUN+="%s --settled --unique --onetime ' $(command -v initqueue)
                     printf -- '--name cryptroot-ask-%%k %s ' $(command -v cryptroot-ask)
-                    printf -- '$env{DEVNAME} luks-$env{ID_FS_UUID} %s"\n' $tout
+                    printf -- '$env{DEVNAME} osencrypt %s"\n' $tout
                 } >> /etc/udev/rules.d/70-luks.rules.new
             else
                 if ! crypttab_contains "$luksid"; then
@@ -43,26 +43,26 @@
 
             uuid=$luksid
             while [ "$uuid" != "${uuid#*-}" ]; do uuid=${uuid%%-*}${uuid#*-}; done
-            printf -- '[ -e /dev/disk/by-id/dm-uuid-CRYPT-LUKS?-*%s*-* ] || exit 1\n' $uuid \
+            printf -- '[ -e /dev/mapper/osencrypt ] || exit 1\n' $uuid \
                 >> $hookdir/initqueue/finished/90-crypt.sh
 
             {
-                printf -- '[ -e /dev/disk/by-uuid/*%s* ] || ' $luksid
+                printf -- '[ -e /dev/mapper/osencrypt ] || ' $luksid
                 printf -- 'warn "crypto LUKS UUID "%s" not found"\n' $luksid
             } >> $hookdir/emergency/90-crypt.sh
         done
     elif getargbool 0 rd.auto; then
-        if [ -z "$DRACUT_SYSTEMD" ]; then
+        if [ 1 ]; then
             {
-                printf -- 'ENV{ID_FS_TYPE}=="crypto_LUKS", RUN+="%s ' $(command -v initqueue)
+                printf -- 'ATTRS{device_id}=="?00000000-0000-*", ATTR{partition}=="ENCRYPTED_DISK_PARTITION", RUN+="%s ' $(command -v initqueue)
                 printf -- '--unique --settled --onetime --name cryptroot-ask-%%k '
-                printf -- '%s $env{DEVNAME} luks-$env{ID_FS_UUID} %s"\n' $(command -v cryptroot-ask) $tout
+                printf -- '%s $env{DEVNAME} osencrypt %s"\n' $(command -v cryptroot-ask) $tout
             } >> /etc/udev/rules.d/70-luks.rules.new
         else
             {
-                printf -- 'ENV{ID_FS_TYPE}=="crypto_LUKS", RUN+="%s ' $(command -v initqueue)
+                printf -- 'ATTRS{device_id}=="?00000000-0000-*", ATTR{partition}=="ENCRYPTED_DISK_PARTITION", RUN+="%s ' $(command -v initqueue)
                 printf -- '--unique --settled --onetime --name crypt-run-generator-%%k '
-                printf -- '%s $env{DEVNAME} luks-$env{ID_FS_UUID}"\n' $(command -v crypt-run-generator)
+                printf -- '%s $env{DEVNAME} osencrypt"\n' $(command -v crypt-run-generator)
             } >> /etc/udev/rules.d/70-luks.rules.new
         fi
     fi
